mobile_app/trigger_build.py

```diff
@@ -0,0 +1,106 @@
+"""
+Triggers a new app build by making a branch for it.
+"""
+
+import argparse
+import git
+import logging
+import json
+import os
+import sys
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+CONFIG_FILE = "CONFIGURATION"
+EXPECTED_ENVIRONMENT_VARIABLES = [
+    "CONFIG_REPO",
+    "CONFIG_PATH",
+    "CONFIG_REVISION",
+    "CODE_REPO",
+    "CODE_REVISION",
+]
+
+
+class MissingEnvironmentVariable(Exception):
+    """
+    Indicates that an expected environment variable was not found
+    """
+    def __init__(self, variable):
+        self.variable = variable
+        super(MissingEnvironmentVariable, self).__init__()
+
+    def __str__(self):
+        return "Missing environment variable: {variable}".format(
+            variable=self.variable
+        )
+
+
+def _build_parser():
+    """
+        Constructs a command line option parser
+
+        Returns:
+            ArgumentParser: The option parser
+    """
+    parser = argparse.ArgumentParser(
+        description="""
+        Commits a list of environment variables to trigger a new build
+        in that environment
+        """
+    )
+    parser.add_argument(
+        '--branch-name',
+        nargs='?',
```

Comment:
As best I can tell, if you use '1' it returns a list. Using '?' returns a single item, which is what I want.